src/vsr/replica.zig

```diff
@@ -6295,6 +6611,24 @@ const DVCQuorum = struct {
         return array;
     }
 
+    fn op_checkpoint_max(dvc_quorum: QuorumMessages) u64 {
+        var checkpoint_max: ?u64 = null;
+        const dvcs = dvcs_all(dvc_quorum);
+        for (dvcs.constSlice()) |dvc| {
+            const dvc_checkpoint = dvc.header.timestamp;
+            if (checkpoint_max == null or checkpoint_max.? < dvc_checkpoint) {
+                checkpoint_max = dvc_checkpoint;
+            }
+        }
+        return checkpoint_max.?;
+    }
+
+    /// Returns the highest op of any header from any canonical-view DVC.
+    /// Includes ops that will be truncated due to gaps.
+    fn op_canonical_max(dvc_quorum: QuorumMessages) u64 {
```

Comment:
Note that we also have `op_max_canonical`, so we probably should rename at least one of them.

Reply:
Oof I forgot about that, how confusing :facepalm:

Reply:
I inlined `op_canonical_max` — it was probably going to be removed as part of handling corrupt headers in DVCs (future PR) anyhow.